imap-simple/imap-simple.d.ts

```diff
@@ -0,0 +1,69 @@
+// Type definitions for imap-simple v1.6.3
+// Project: https://github.com/chadxz/imap-simple
+// Definitions by: Jeffery Grajkowski <https://github.com/pushplay/>
+// Definitions: https://github.com/psnider/DefinitelyTyped/imap-simple
+
+/// <reference path="../imap/imap.d.ts" />
+
+declare namespace IMAPS {
+
+    export interface ImapSimpleOptions {
+        /** Options to pass to node-imap constructor. */
+        imap: IMAP.Config;
+
+        /** Time in milliseconds to wait before giving up on a connection attempt. (Deprecated: please use options.imap.authTimeout instead) */
+        connectTimeout?: number;
+    }
+
+    export interface MessageBodyPart extends IMAP.ImapMessageBodyInfo {
+        /** string type where which=='TEXT', complex Object where which=='HEADER' */
+        body: any;
+    }
+
+    export interface Message {
+        attributes: IMAP.ImapMessageAttributes;
+        parts: MessageBodyPart[];
+    }
+
+    export class ImapSimple {
+        constructor(imap: IMAP.Connection);
+
+        /** Open a mailbox, calling the provided callback with signature (err, boxName), or resolves the returned promise with boxName. */
+        openBox(boxName: string, callback?: (err: Error, boxName: string) => void): Promise<string>;
```

Comment:
I thought both callback and Promise would work together but I just tested and you're right, it's one or the other.